multiple-servers/api/images/images.go

```diff
@@ -0,0 +1,48 @@
+package images
+
+import (
+	"context"
+	. "multiple-servers/api/types"
+
+	"github.com/jackc/pgx/v4"
+)
+
+func GetAll(conn *pgx.Conn) ([]Image, error) {
+	rows, err := conn.Query(context.Background(), "SELECT title, alt_text, url FROM images")
+	if err != nil {
+		return nil, err
+	}
+
+	var images []Image
+
+	for rows.Next() {
+		var image Image
+		err = rows.Scan(&image.Title, &image.AltText, &image.Url)
+		if err != nil {
+			return images, err
+		}
+		images = append(images, image)
+	}
+
+	return images, nil
+}
+
+func InsertOne(conn *pgx.Conn, newImage Image) (Image, error) {
```

Comment:
I thought returning what was created is a good idea, but I suppose the status 201 should be sufficient

Reply:
I was more asking why the _function_ returns the `Image`, rather than the HTTP endpoint.

As a function, if the caller has exactly the information I'm going to return to them, it doesn't often make sense to return it - they already _have_ it.

```go
image := ...
image2 := InsertOne(image)
// Why would I use `image2` here if `image` and `image2` are identical? I could just use `image`
```

There are cases where it may make sense (e.g. if you're filling in an ID, or if you're doing some kind of fluent method chaining), but I'm not sure it makes sense here.